src/components/Visualizer.vue

```diff
@@ -103,23 +79,32 @@ export default {
         this.visualizer = this.visualizer == this.visualizerArr.length - 1 ? 0 : this.visualizer + 1;
     },
     update() {
-        if(!this.autoUpdate) return;
+        if(!this.autoUpdate || !this.ctx) return;
         requestAnimationFrame(this.update.bind(this));
         this.renderVisualizer();
     }
   },
     watch : {
-        audio: function(){
-            this.initAllVisualizersIfRequried()
+        audioData: function(){
```

Comment:
```suggestion
        audioData: () => {
```

Favour arrow function over anonymous function where possible since you don't need `this` binding.

Reply:
We should setup an ESLint rule for this.

Reply:
Sure Indeed